### PR TITLE
Add convenience functions for pwhash() and derive_key()

### PR DESCRIPTION
**Pull request by manuels**
The limit arguments in pwhash() and derive_key() with its two parameters is a bit "loud" with its two (for most users) constant parameters. This adds two convenience functions for both.

### CODE REVIEW

File: src/crypto/pwhash/scryptsalsa208sha256.rs [hunk 1]
```diff
@@ -124,6 +124,28 @@ pub fn derive_key<'a>(
     }
 }
 
+/// `derive_key_interactive()` is a shortcut function for `derive_key()` with
+/// interactive limits (i.e. using `derive_key()` with `OPSLIMIT_INTERACTIVE`
+/// and `MEMLIMIT_INTERACTIVE`)
+pub fn derive_key_interactive<'a>(
```
Comment (kpp) [inline src/crypto/pwhash/scryptsalsa208sha256.rs]: Do you need that `<'a>`?

Comment (manuels) [reply src/crypto/pwhash/scryptsalsa208sha256.rs]: Yes, the compiler will fail if `'a` is missing.

**Comment by Dylan-DPC** [comment]
bors: try

**Comment by manuels** [comment]
bors: try

**Comment by Dylan-DPC** [comment]
bors: r+

**Pull request merged**